crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -252,6 +363,133 @@ fn check_class_attribute(checker: &mut Checker, attribute_expr: &ExprAttribute)
     }
 }
 
+/// Finds the parameter definition for a given name expression in a function.
+fn find_parameter<'a>(
+    semantic: &'a SemanticModel,
+    name: &'a ExprName,
+) -> Option<AnyParameterRef<'a>> {
+    let binding_id = semantic.only_binding(name)?;
+    let binding = semantic.binding(binding_id);
+    let StmtFunctionDef { parameters, .. } = binding.statement(semantic)?.as_function_def_stmt()?;
+    parameters
+        .iter()
+        .find(|parameter| parameter.name().range() == binding.range())
+}
+
+/// Checks whether an Airflow 3.0–removed context key is used in a function decorated with `@task`.
+///
+/// Specifically, it flags two scenarios for task decorated function:
+/// 1. A removed context variable passed in as a function parameter name (e.g., `execution_date`).
+/// 2. A removed context key accessed via `context.get("...")`.
+///
+/// # Examples
+///
+/// **Removed key used in `context.get(...)`:**
+/// ```python
+/// from airflow.decorators import task
+///
+/// @task
+/// def my_task(**context):
+///     # 'conf' is removed in Airflow 3.0
+///     print(context.get("conf"))
+/// ```
+///
+/// **Accessing multiple keys:**
+/// ```python
+/// from airflow.decorators import task
+///
+/// @task
+/// def more_keys(**context):
+///     # 'prev_ds' is also removed in Airflow 3.0
+///     print(context.get("prev_ds"))
+/// ```
+fn check_removed_context_keys_usage(checker: &mut Checker, call_expr: &ExprCall) {
+    if !is_taskflow(checker) {
+        return;
+    }
+
+    let Expr::Attribute(ExprAttribute { value, attr, .. }) = &*call_expr.func else {
+        return;
+    };
+
+    let is_named_context = if let Expr::Name(name) = &**value {
+        if let Some(parameter) = find_parameter(checker.semantic(), name) {
+            matches!(parameter.name().as_str(), "context" | "kwargs")
+                || parameter.name().as_str().starts_with("**")
```

Comment:
The parameter name won't really contain `**` at the start. You can see that in the playground: https://play.ruff.rs/f7051131-c6c0-49a4-8b91-e74c37ae8088